src/runtime/runtime.ts

```diff
@@ -1,20 +1,32 @@
+//------------------------------------------------------------------------
+// Runtime
+//------------------------------------------------------------------------
+
 type ID = number;
 type Multiplicity = number;
 
 function createHash() {
   return Object.create(null);
 }
 
+function createArray() {
```

Comment:
What's the idea behind `createArray`? Consistency with `createHash`, pretty name or something else entirely?